bin/packages/build-worker.js

```diff
@@ -76,109 +76,114 @@ function getBuildPath( file, buildFolder ) {
 		);
-		const outputFileRTL = getBuildPath(
-			file.replace( '.scss', '-rtl.css' ),
-			'build-style'
+		const babelOptions = getBabelConfig(
+			environment,
+			file.replace( PACKAGES_DIR, '@wordpress' )
 		);
 
-		const [ , contents ] = await Promise.all( [
-			makeDir( path.dirname( outputFile ) ),
-			readFile( file, 'utf8' ),
+		const [ , transformed ] = await Promise.all( [
+			makeDir( path.dirname( destPath ) ),
+			babel.transformFileAsync( file, babelOptions ),
 		] );
-		const builtSass = await renderSass( {
-			file,
-			includePaths: [ path.join( PACKAGES_DIR, 'base-styles' ) ],
-			data:
-				[
-					'colors',
-					'breakpoints',
-					'variables',
-					'mixins',
-					'animations',
-					'z-index',
-				]
-					// Editor styles should be excluded from the default CSS vars output.
-					.concat(
-						file.includes( 'common.scss' ) ||
-							( ! file.includes( 'block-library' ) &&
-								! file.includes( 'editor-styles.scss' ) )
-							? [ 'default-custom-properties' ]
-							: []
-					)
-					.map( ( imported ) => `@import "${ imported }";` )
-					.join( ' ' ) + contents,
-		} );
-
-		const result = await postcss(
-			require( '@wordpress/postcss-plugins-preset' )
-		).process( builtSass.css, {
-			from: 'src/app.css',
-			to: 'dest/app.css',
-		} );
-
-		const resultRTL = await postcss( [ require( 'rtlcss' )() ] ).process(
-			result.css,
-			{
-				from: 'src/app.css',
-				to: 'dest/app.css',
-			}
-		);
 
 		await Promise.all( [
-			writeFile( outputFile, result.css ),
-			writeFile( outputFileRTL, resultRTL.css ),
+			writeFile( destPath + '.map', JSON.stringify( transformed.map ) ),
+			writeFile(
+				destPath,
+				transformed.code +
+					'\n//# sourceMappingURL=' +
+					path.basename( destPath ) +
+					'.map'
+			),
 		] );
-	},
-
-	async '.js'( file ) {
-		for ( const [ environment, buildDir ] of Object.entries(
-			JS_ENVIRONMENTS
-		) ) {
-			const destPath = getBuildPath( file, buildDir );
-			const babelOptions = getBabelConfig(
-				environment,
-				file.replace( PACKAGES_DIR, '@wordpress' )
-			);
-
-			const [ , transformed ] = await Promise.all( [
-				makeDir( path.dirname( destPath ) ),
-				babel.transformFileAsync( file, babelOptions ),
-			] );
-
-			await Promise.all( [
-				writeFile(
-					destPath + '.map',
-					JSON.stringify( transformed.map )
-				),
-				writeFile(
-					destPath,
-					transformed.code +
-						'\n//# sourceMappingURL=' +
-						path.basename( destPath ) +
-						'.map'
-				),
-			] );
-		}
-	},
+	}
+}
+
+/**
+ * Object of build tasks per file extension.
+ *
+ * @type {Object<string,Function>}
+ */
+const BUILD_TASK_BY_EXTENSION = {
+	'.scss': buildCSS,
+	'.js': buildJS,
+	'.ts': buildJS,
+	'.tsx': buildJS,
 };
 
 module.exports = async ( file, callback ) => {
 	const extension = path.extname( file );
 	const task = BUILD_TASK_BY_EXTENSION[ extension ];
 
 	if ( ! task ) {
-		return;
+		callback( new Error( `No handler for extension: ${ extension }` ) );
```

Comment:
👍  Nice, expose the problem.

Reply:
Without this, the `callback` was never called and the build script was stuck forever.